substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/HeapImpl.java

```diff
@@ -118,6 +119,10 @@ public final class HeapImpl extends Heap {
     /** A cached list of all the classes, if someone asks for it. */
     private List<Class<?>> classList;
 
+    /** A cached list of all the loaded classes, if someone asks for it. */
+    private List<Class<?>> loadedClasses;
```

Comment:
I don't know if this will work at all, but did you consider using `PredefinedClassesSupport.loadedClassesByName` instead of creating a new cache?

Reply:
Hmm that's a good idea.  I gave it a try, but I think that `PredefinedClassesSupport.loadedClassesByName` doesn't actually cache **all** loaded classes, just the ones that are "dynamically" loaded at runtime (although they aren't truly "loaded" in the usual sense).
If I try to reuse `PredefinedClassesSupport.loadedClassesByName`, most of the classes show up as `null`. See below: 

![image](https://github.com/oracle/graal/assets/35396843/577b81b3-f7b9-4148-9c30-09ecbbe78bff)